ibtk/src/math/HierarchyMathOps.cpp

```diff
@@ -1229,6 +1229,89 @@ HierarchyMathOps::div(const int dst_idx,
     return;
 } // div
 
+void
+HierarchyMathOps::div(const int dst_idx,
+                      const Pointer<CellVariable<NDIM, double> > dst_var,
+                      const double alpha,
+                      const int src1_idx,
+                      const Pointer<SideVariable<NDIM, double> > src1_var,
+                      const Pointer<HierarchyGhostCellInterpolation> src1_ghost_fill,
+                      const double src1_ghost_fill_time,
+                      const bool src1_cf_bdry_synch,
+                      const int coef1_idx,
+                      const Pointer<SideVariable<NDIM, double> > /*coef1_var*/,
+                      const Pointer<HierarchyGhostCellInterpolation> coef1_ghost_fill,
+                      const double coef1_ghost_fill_time,
+                      const double beta,
+                      const int src2_idx,
+                      const Pointer<CellVariable<NDIM, double> > src2_var,
+                      const int dst_depth,
+                      const int src2_depth)
+{
+    if (coef1_idx < 0)
```

Comment:
OK, now I see why `alpha` and `coef1` are split up in the interface.

1. I think that I would rather move this logic out of `HierarchyMathOps` and make it the caller's job to know which `div` implementation they want to use. Are there situations where having `HierarchyMathOps` handle this internally makes a big simplification in algorithm implementation?
2. Instead of `(coef1_idx < 0)`, prefer to use `invalid_index`:
```suggestion
    if (coef1_idx == invalid_index)
```